plugins/inputs/nginx_plus/nginx_plus.go

```diff
@@ -0,0 +1,531 @@
+package nginx_plus
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"net"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+type NginxPlus struct {
+	Urls []string
+}
+
+var sampleConfig = `
+  ## An array of ngx_http_status_module or status URI to gather stats.
+  urls = ["http://localhost/status"]
+`
+
+func (n *NginxPlus) SampleConfig() string {
+	return sampleConfig
+}
+
+func (n *NginxPlus) Description() string {
+	return "Read Nginx Plus' full status information (ngx_http_status_module)"
+}
+
+func (n *NginxPlus) Gather(acc telegraf.Accumulator) error {
+	var wg sync.WaitGroup
+
+	for _, u := range n.Urls {
+		addr, err := url.Parse(u)
+		if err != nil {
+			acc.AddError(fmt.Errorf("Unable to parse address '%s': %s", u, err))
+		}
+
+		wg.Add(1)
+		go func(addr *url.URL) {
+			defer wg.Done()
+			acc.AddError(n.gatherUrl(addr, acc))
+		}(addr)
+	}
+
+	wg.Wait()
+	return nil
+}
+
+var tr = &http.Transport{
+	ResponseHeaderTimeout: time.Duration(3 * time.Second),
+}
+
+var client = &http.Client{
+	Transport: tr,
+	Timeout:   time.Duration(4 * time.Second),
+}
```

Comment:
Make the Client and Transport part of the NginxPlus struct, this way they won't need to share configuration values if you have more than one plugin instance.

Bonus points if you want to add SSL support and a `timeout` option, but not required.